### PR TITLE
CRYP-37: As a User, I want to filter my Ethereum wallet transactions by tag, so that I can search for specific transaction characteristics

### PR DESCRIPTION
**Pull request by alexbara2000**
image

image

image

### CODE REVIEW

File: packages/client/screens/filters/FilterScreen.tsx [hunk 1]
```diff
@@ -180,20 +188,82 @@ export default function FilterScreen({ route, navigation }: HomeStackScreenProps
...
+                            <HStack space={"10px"} alignItems={"center"}>
+                                <Text size={"subheadline"} fontWeight={"semibold"}>
+                                    {tag}
+                                </Text>
+                            </HStack>
+                        </Pressable>
+                    ))}
+                    <FontAwesomeIcon icon={farChevronRight} style={styles.chevronRightIcon} size={16} />
+                </HStack>
+            </Pressable>
```
Comment (jason-gerard) [inline packages/client/screens/filters/FilterScreen.tsx:222]: Why is this duplicated

File: packages/client/screens/filters/FilterScreen.tsx [hunk 2]
```diff
@@ -180,20 +188,82 @@ export default function FilterScreen({ route, navigation }: HomeStackScreenProps
...
                 ) &&
                     (!isFilterSaved ||
                         !(
                             filterByTransaction === route.params.filters[0] &&
                             filterByDate === route.params.filters[1] &&
-                            filterByContact.length === 0
-                        )) && (
+                            filterByContact.length === 0 &&
+                        filterByTag.length === 0&&
+                        filterByTag.length === 0)) && (
```
Comment (jason-gerard) [inline packages/client/screens/filters/FilterScreen.tsx:266]: and this also?

File: packages/client/screens/filters/FilterScreen.tsx [hunk 3]
```diff
@@ -264,4 +335,14 @@ const styles = StyleSheet.create({
         marginLeft: "auto",
         marginRight: 5,
     },
+    tags: {
+        marginBottom: -13,
+    },
```
Comment (jason-gerard) [inline packages/client/screens/filters/FilterScreen.tsx:340]: What is this for?

**Review by jason-gerard — COMMENTED** [review]
Good start, seems pretty messy. Confused by the duplicate code... Any reason for this?

**Comment by jason-gerard** [comment]
You can run the linter with the command `yarn run lint:fix` in the root of the project

File: packages/client/screens/filters/FilterScreen.tsx (hunk 1 above)
Comment (alexbara2000) [reply packages/client/screens/filters/FilterScreen.tsx]: good question, I think it is the rebase

**Review by jason-gerard — APPROVED** [review]

**Review by Andre-Ibrahim — APPROVED** [review]
To micro optimize the cli 's runtime you can do `yarn lint:fix` instead of `yarn run lint:fix`. This reduces the amount of characters the CLI needs to parse.

**Comment by alexbara2000** [comment]
Thank you for the input. Further micro optimization can also be done if you run "yarn" instead of "yarn install". But that brings us to quantum computer science so it gets tricky.

**Pull request merged**